src/test/net_tests.cpp

```diff
@@ -319,4 +321,73 @@ BOOST_AUTO_TEST_CASE(PoissonNextSend)
     g_mock_deterministic_tests = false;
 }
 
+void message_serialize_deserialize_test(bool v2, const std::vector<CSerializedNetMsg>& test_msgs) {
+    // use 32 bytey keys with all zeros
+    CPrivKey k1(32, 0);
+    CPrivKey k2(32, 0);
+
+    // construct the serializers
+    std::unique_ptr<TransportSerializer> serializer;
+    std::unique_ptr<TransportDeserializer> deserializer;
+
+    if (v2) {
+        serializer = MakeUnique<V2TransportSerializer>(V2TransportSerializer(k1, k2));
+        deserializer = MakeUnique<V2TransportDeserializer>(V2TransportDeserializer(Params().MessageStart(), k1, k2));
+    }
+    else {
```

Comment:
I could be wrong but style guide seems to say these should be on the same line https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#coding-style-c
```suggestion
    } else {
```